### DISCUSSION TITLE
Username/Password + Public key authentication

### DISCUSSION THREAD

**Discussion by gary-patel-sonder** [drakkan/sftpgo#1281] [General]
Is it possible to require both the username/password + public key to connect to the SFTP server? Or is it one or the other?

**Comment by drakkan** [comment]
Hello, multi-step authentication is supported. I no longer provide free step-by-step support, sorry, please check the [support policy](https://github.com/drakkan/sftpgo#support-policy). Hope you understand. Thank you

**Reply by gary-patel-sonder** [reply]
@drakkan can you confirm the person connecting to the sftp would be required to use their username/password + private key to connect and not MFA. If so, I will go ahead and purchase the paid version for support.

**Reply by drakkan** [reply]
> @drakkan can you confirm the person connecting to the sftp would be required to use their username/password + private key to connect and not MFA. If so, I will go ahead and purchase the paid version for support.

Thank you for considering supporting the project.

Both are supported:

- multi step authentication: key first and then password
- multi factor authentication: password and then one time token
